src/mono/mono/metadata/object.c

```diff
@@ -5073,6 +5081,34 @@ mono_unhandled_exception_checked (MonoObjectHandle exc, MonoError *error)
 			call_unhandled_exception_delegate (current_domain, current_appdomain_delegate, exc);
 		mono_threads_end_abort_protected_block ();
 	}
+#else
+	MonoObject *delegate = NULL;
+	MonoObjectHandle delegate_handle;
+	MonoVTable *vt = mono_class_vtable_checked (current_domain, mono_defaults.appcontext_class, error);
+	goto_if_nok (error, leave);
+
+	// TODO: use handles directly
+	mono_field_static_get_value_checked (vt, field, &delegate, MONO_HANDLE_NEW (MonoString, NULL), error);
+	goto_if_nok (error, leave);
+	delegate_handle = MONO_HANDLE_NEW (MonoObject, delegate);
+
+	// Unhandled exception callbacks must not be aborted
+	mono_threads_begin_abort_protected_block ();
```

Comment:
Not sure we need these. netcore doesn't have Thread.Abort.

Reply:
Removed in a followup commit after we chatted on Slack.